amundsen_application/static/js/ducks/search/sagas.ts

```diff
@@ -53,7 +76,90 @@ export function* searchResourceWorker(action: SearchResourceRequest): SagaIterat
   } catch (e) {
     yield put(searchResourceFailure());
   }
-}
+};
 export function* searchResourceWatcher(): SagaIterator {
   yield takeEvery(SearchResource.REQUEST, searchResourceWorker);
-}
+};
+
+export function* submitSearchWorker(action: SubmitSearchRequest): SagaIterator {
+  const { searchTerm } = action.payload;
+  yield put(searchAll(searchTerm, ResourceType.table, 0));
```

Comment:
This should account for if `searchAll` fails.

Reply:
Discussed offline. The `yield put(searchAll` simply initiates another saga, and the failure case is handled in the `searchAll` saga.